lib/AIECreatePathfindFlows.cpp

```diff
@@ -207,10 +207,13 @@ class DynamicTileAnalysis {
     if(coordToShimMux.count(std::make_pair(col, row))) {
       return coordToShimMux[std::make_pair(col, row)];
     } else {
-      assert(getTile(builder, col, row).isShimNOCTile());
       ShimMuxOp switchboxOp =
         builder.create<ShimMuxOp>(builder.getUnknownLoc(),
                                     getTile(builder, col, row));
+      // check if the shim tile has noc connection
+      if (!getTile(builder, col, row).isShimNOCTile())
+        switchboxOp.emitError("Attempting to route to or from ShimTile (")
+            << col << ", " << row << "), which has no NOC connection";
```

Comment:
So this is still somewhat problematic, because the ShimMuxOp is not really visible to the user.  Maybe it should emit the error on the FlowOp being routed and then have  a note that points to the shimmux or tile op where the error was encountered?

Reply:
I agree with this - is there a way to put the error on the flow-op that failed.

Reply:
Agreed. This test has now been moved to where the src and dst TileOps of every FlowOp are first created. Also linked this error to the FlowOp instead of ShimMuxOp.